hack/ldapsearch.sh

```diff
@@ -0,0 +1,223 @@
+    if [[ "$#" == "0" || "$1" == -* ]]; then
+      log_error "-n|--namespace requires a namespace name to be specified"
+      exit 1
+    fi
+    namespace=$1
+    shift
+    ;;
+  -u | --username)
+    shift
+    # If there are no more command line arguments, or there is another command line argument but it starts with a dash, then error
+    if [[ "$#" == "0" || "$1" == -* ]]; then
+      log_error "-u|--username requires a username to be specified"
+      exit 1
+    fi
+    username=$1
+    shift
+    ;;
+  -*)
+    log_error "Unsupported flag $1" >&2
+    if [[ "$1" == *"active-directory"* ]]; then
+      log_error "Did you mean --get-active-directory-vars?"
+    fi
+    exit 1
+    ;;
+  *)
+    log_error "Unsupported positional arg $1" >&2
+    exit 1
+    ;;
+  esac
+done
+
+if [[ "$help" == "yes" ]]; then
+  me="$(basename "${BASH_SOURCE[0]}")"
+  log_note "Usage:"
+  log_note "   $me [flags]"
+  log_note
+  log_note "Flags:"
+  log_note "   -h, --help:      print this usage"
+  log_note "   -r, --resource:  specify the resource type and name (e.g. ldapidentityprovider/my-ldap-idp) - required"
+  log_note "   -n, --namespace: specify the namespace in which the resource exists - required"
+  log_note "   -u, --username:  specify a username, as an end-user would type it during a login - required"
+  exit 1
+fi
+
+if [[ -z "$namespace" ]]; then
+  log_error "-n|--namespace is required"
+  exit 1
+fi
+
+if [[ -z "$resource_type_and_name" ]]; then
+  log_error "-r|--resource is required"
+  exit 1
+fi
+
+if [[ "$resource_type_and_name" != ldapidentityprovider/* ]]; then
+  log_error "-r|--resource currently only supports ldapidentityprovider type resources."
+  log_error "Please specify the value as \"ldapidentityprovider/name-of-resource\"."
+  exit 1
+fi
+
+if [[ -z "$username" ]]; then
+  log_error "-u|--username is required"
+  exit 1
+fi
+
+RESOURCE_FILE=$(mktemp)
+trap "rm $RESOURCE_FILE" EXIT
+
+kubectl get "$resource_type_and_name" \
+  --namespace "$namespace" \
+  --output yaml >"$RESOURCE_FILE"
+
+# See docs for LDAPIdentityProvider.spec for details about these settings.
+# https://github.com/vmware-tanzu/pinniped/blob/main/generated/1.28/README.adoc#k8s-api-go-pinniped-dev-generated-1-28-apis-supervisor-idp-v1alpha1-ldapidentityproviderspec
+LDAP_HOST=$(yq '.spec.host' "$RESOURCE_FILE")                                                                       # required
+LDAP_CA_BUNDLE=$(yq '.spec.tls.certificateAuthorityData // ""' "$RESOURCE_FILE")                                    # optional
+LDAP_BIND_SECRETNAME=$(yq '.spec.bind.secretName' "$RESOURCE_FILE")                                                 # required
+LDAP_USER_SEARCH_BASE=$(yq '.spec.userSearch.base' "$RESOURCE_FILE")                                                # required
+LDAP_USER_SEARCH_FILTER=$(yq '.spec.userSearch.filter // ""' "$RESOURCE_FILE")                                      # optional unless Attributes.Username is set to "dn"
```

Comment:
We need to add a check here to wrap this in `()` if it isn't already.

Reply:
Done!